src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -148,6 +147,23 @@ public function CollectExternalResources($sPageContent)
         return $sPageContent;
     }
 
+    private function getFilesRefreshAndDomainPrefix(): string
+    {
+        $filesPrefix = ServiceLocator::getParameter('chameleon_system_core.resources.enable_external_resource_collection_refresh_prefix');
+
+        /**
+         * @var $portalDomainService PortalDomainServiceInterface
+         */
+        $portalDomainService = ServiceLocator::get('chameleon_system_core.portal_domain_service');
```

Comment:
Could you move the ServiceLocator call to a separate method that includes a return type declaration? This improves type safety and keeps the distracting details of service retrieval away.

Reply:
ok